2021-01-29 11:13:49.878 +07:00 [INF] Starting web host
2021-01-29 11:30:16.903 +07:00 [INF] Starting web host
2021-01-29 11:31:12.433 +07:00 [INF] Starting web host
2021-01-29 11:31:34.086 +07:00 [INF] Start Add Character process.
2021-01-29 11:31:35.033 +07:00 [INF] End.
2021-01-29 11:37:13.412 +07:00 [INF] Starting web host
2021-01-29 11:38:18.144 +07:00 [INF] Start Add Character process.
2021-01-29 11:38:19.138 +07:00 [ERR] Duplicated Character Name.
2021-01-29 11:38:38.096 +07:00 [INF] Start Add Character process.
2021-01-29 11:38:38.129 +07:00 [INF] Add New Character.
2021-01-29 11:38:38.298 +07:00 [INF] Success.
2021-01-29 11:38:38.316 +07:00 [INF] End.
2021-01-29 12:15:07.998 +07:00 [INF] Starting web host
2021-01-29 13:43:58.655 +07:00 [INF] Starting web host
2021-01-29 14:07:41.090 +07:00 [INF] Starting web host
2021-01-29 14:08:39.815 +07:00 [INF] Starting web host
2021-01-29 14:10:24.066 +07:00 [ERR] This attackerId 9 not found.
2021-01-29 14:10:30.691 +07:00 [ERR] Object reference not set to an instance of an object.
2021-01-29 14:10:54.627 +07:00 [ERR] Object reference not set to an instance of an object.
2021-01-29 14:14:10.194 +07:00 [INF] Starting web host
2021-01-29 14:38:05.087 +07:00 [INF] Starting web host
2021-01-29 15:01:59.654 +07:00 [INF] Starting web host
2021-01-29 16:14:08.043 +07:00 [INF] Starting web host
2021-01-29 16:17:18.973 +07:00 [INF] Starting web host
2021-01-29 16:17:27.804 +07:00 [ERR] An unhandled exception has occurred while executing the request.
System.InvalidOperationException: Unable to resolve service for type 'RPG_Project.Services.Product.IProductService' while attempting to activate 'RPG_Project.Controllers.ProductController'.
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.GetService(IServiceProvider sp, Type type, Type requiredBy, Boolean isDefaultParameterRequired)
   at lambda_method(Closure , IServiceProvider , Object[] )
   at Microsoft.AspNetCore.Mvc.Controllers.ControllerActivatorProvider.<>c__DisplayClass4_0.<CreateActivator>b__0(ControllerContext controllerContext)
   at Microsoft.AspNetCore.Mvc.Controllers.ControllerFactoryProvider.<>c__DisplayClass5_0.<CreateControllerFactory>g__CreateController|0(ControllerContext controllerContext)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.ResponseCaching.ResponseCachingMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
